src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,154 @@
+                if (descriptionExists(inputArr, "Please specify a task number!")) {
+                    int taskNum = Integer.parseInt(description);
+                    return new DeleteCommand(taskNum);
+                }
+                break;
+            case "mark":
+                if (descriptionExists(inputArr, "Please specify a task number!")) {
+                    int taskNum = Integer.parseInt(description);
+                    return new MarkCommand(taskNum);
+                }
+                break;
+            case "unmark":
+                if (descriptionExists(inputArr, "Please specify a task number!")) {
+                    int taskNum = Integer.parseInt(description);
+                    return new UnmarkCommand(taskNum);
+                }
+                break;
+            case "todo":
+                if (descriptionExists(inputArr, "Oops, a todo description cannot be left empty!")) {
+                    Task task = new Todo(description);
+                    return new AddCommand(task);
+                }
+                break;
+            case "deadline":
+                if (descriptionExists(inputArr, "Oops, a deadline description cannot be left empty!")) {
+                    String[] descrArr = description.split(" /by ");
+                    if (keywordExists(descrArr, "/by", "deadline")) {
+                        LocalDateTime dateTime = parseDateTime(descrArr[1]);
+                        Task task = new Deadline(descrArr[0], dateTime);
+                        return new AddCommand(task);
+                    }
+                }
+                break;
+            case "event":
+                if (descriptionExists(inputArr, "Oops, an event description cannot be left empty!")) {
+                    String[] descrArr = description.split(" /at ");
+                    if (keywordExists(descrArr, "/at", "event")) {
+                        LocalDateTime dateTime = parseDateTime(descrArr[1]);
+                        Task task = new Event(descrArr[0], dateTime);
+                        return new AddCommand(task);
+                    }
+                }
+                break;
+            case "find":
+                if (descriptionExists(inputArr, "Please specify a keyword!")) {
+                    if (inputArr.length > 2) {
+                        throw new DukeException("Please specify only one keyword!");
+                    } else {
+                        return new FindCommand(description);
+                    }
+                }
+                break;
+            case "bye":
+                return new ExitCommand();
+            default:
+                throw new DukeException("Oh no! I don't understand what that means...");
+            }
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Invalid date/time format.\n    "
+                    + "Please use the following format: yyyy-mm-dd HH:mm");
+        }
+        return new InvalidCommand();
```

Comment:
Not sure if I'm understanding this right, but is this line ever reached? I feel like the default clause of the switch-case statement already handles invalid commands.